server/database/repositories/TournamentRepository.ts

```diff
@@ -0,0 +1,45 @@
+import { Document } from 'mongoose';
+import TournamentModel from '../models/TournamentModel';
+import Tournament, { Team } from '../../../shared/types/Tournament';
+import TeamModel from '../models/TeamModel';
+
+export default class TournamentRepository {
+  public static insert = async (tournament: Tournament) => {
+    const document = new TournamentModel(tournament);
+    let savedTournament : Tournament;
+    try {
+      savedTournament = await document.save();
+    } catch (err) {
+      throw new Error(err);
+    }
+    return savedTournament;
+  };
+
+  public static insertTeam = async (tournament: Tournament | string, team : Team | string) => {
+    let tournamentDoc : (Document & Tournament) | null;
```

Comment:
```suggestion
    let tournamentDoc: TournamentModel;
```
I'm not sure, but it could work